### PR TITLE
Hakawai functionality

### PR DESCRIPTION
**Pull request by simonsmiley64**
This pull request adds on the following:

- Added Hakawai pod to the podspec and substitution of the textView with the Hakawai textView. This makes it easier for us to add the mention functionality as a module and probably gives us more power with the TextView
- Message Meta data: I have added and modified the message to allow us to add meta data with it. I have tried to leave all the functions the same and link them in with the current send text functions. We have to add a dictionary as meta (even if empty). We can now therefore choose to update meta with messages when they are sent.

- I am not sure why the StickerMessage.readMe stuff snuck into the pull request, I am still getting used to manipulating git effectively

### CODE REVIEW

File: ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m [hunk 1]
```diff
@@ -99,7 +99,8 @@ -(NSDictionary *) serialize {
              b_Type: _model.type,
              b_Date: [FIRServerValue timestamp],
              b_UserFirebaseID: _model.userModel.entityID,
-             b_ReadPath: self.initialReadReceipts};
+             b_ReadPath: self.initialReadReceipts,
+             b_Meta: _model.metaDictionary};
```
Comment (bensmiley) [inline ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m]: What happens if the meta is Nil?

File: Docs/Modules/StickerMessages.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+## Installation
```
Comment (bensmiley) [inline Docs/Modules/StickerMessages.md]: This file has been deleted in the latest version. Why is it included in this pull request? Could you make sure that you are using the latest version of the code?

File: ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m (hunk 1 above)
Comment (simonsmiley64) [reply ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m]: Meta should never be nil.

Currently we are only dealing with mention meta data, it might be that in the future we add image message meta etc. For functions with meta data we set .metaDictionary as the meta. For all other message types we just set the meta Dictionary as a nil dictionary .metaDictionary = @{}. This lays the foundation for meta data without requiring us to add checks. 

Note: I forgot to add this in this pull request - in further testing I found this to be an issue so have added empty meta dictionaries to the other message types

File: Docs/Modules/StickerMessages.md (hunk 2 above)
Comment (simonsmiley64) [reply Docs/Modules/StickerMessages.md]: Deleted

**Comment by simonsmiley64** [comment]
I have added some fixes to the meta data code. I have tested these fixes and adding them (without any meta data) doesn't have any affect on the regular code.

It should though enable us to add meta data for certain message types

File: ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m (hunk 1 above)
Comment (bensmiley) [inline ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m]: What I meant was that you could just add a check here:

```
b_Meta: _model.metaDictionary ? _model.metaDictionary : @{},
```

File: ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m [hunk 3]
```diff
@@ -162,6 +163,11 @@ -(RXPromise *) deserialize: (NSDictionary *) value {
         //[_model setReadReceipts:readReceipts];
     }
     
+    NSDictionary * meta = value[bMetaDataPath];
```
Comment (bensmiley) [inline ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m]: Why do you use `b_Meta` as the key above and then `bMetaDataPath` here?

File: ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m (hunk 1 above)
Comment (bensmiley) [reply ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m]: ^^ Review comment above.

File: ChatSDK/ChatSDKUI/Classes/Components/Chat View/BTextInputView.m [hunk 4]
```diff
@@ -68,8 +67,12 @@ - (id)init
...
 
         // Create a text view
-        _textView = [[UITextView alloc] init];
-        _textView.delegate = self;
+        _textView = [[HKWTextView alloc] init];
+        
+        // If we use the mentions functionality we need to set the external delegate
+        // This is the way to set the UITextView delegate to keep mentions functionality working
+        _textView.simpleDelegate = self;
+
```
Comment (bensmiley) [inline ChatSDK/ChatSDKUI/Classes/Components/Chat View/BTextInputView.m]: Is the simple delegate a super set of the standard delegate method? If we get rid of the delegate, will we still get callbacks when text is typed etc...? Have you tested the mentions functionality with multiple lines?

Comment (simonsmiley64) [reply ChatSDK/ChatSDKUI/Classes/Components/Chat View/BTextInputView.m]: The HKWTextView has two delegates, one for the advanced functionality and one for the regular textView functionality. 

This means we need to set the simpleDelegate here to ensure all the typing functions are called. (You can test this by putting a break point in shouldChangeTextInRange). Later in the mention view we call the actual delegate which then connects it to all the mention side of things. 

Setting the simple delegate here allows the HKWTextView to act as a regular textView until we add the mentions module and then it gets upgraded.

File: ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m (hunk 1 above)
Comment (simonsmiley64) [reply ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m]: Fixed

File: ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m (hunk 3 above)
Comment (simonsmiley64) [reply ChatSDK/ChatSDKFirebaseAdapter/Classes/Firebase/CCMessageWrapper.m]: Fixed
